### PR TITLE
Add tck tests for Message Payload various formats

### LINKED ISSUE #11: Add tests for all schema formats
There are no TCK tests for JSON Schema draft7. Although AsyncAPI default schema format is a superset of JSON Schema draft7, it would be good to have tests that use external JSON Schema draft7 documents the way a lot of people would normally use them. 

Additionally, it would be nice to have TCK tests for all the recommended schema formats mentioned in the spec as well, namely: Avro, OAS 3.0 schema, and RAML 1.0 DataType. However, for we may want to make such test informational/optional so as to not lower the percentage coverage for something that is not mandated by the spec.

A few sample use cases to test: external JSON Schema draft7,  $ref-erenced schema component within and OAS 3.0 document, $ref-erenced RAML Datatype (standalone), $ref-erenced RAML Datatype (in RAML Library).

### PR DESCRIPTION
**Pull request by postatum**
**Description**
Adds tck tests of using external documents of various formats in Message Payload:
- JSON Schema JSON and YAML (duplicate of https://github.com/asyncapi/tck/pull/13)
- RAML DataType
- Avro Schema JSON and YAML

It's currently not possible to mark tck tests as optional so all of the new tests affect the percentage coverage.

Also I didn't add RAML Library tests because AsyncAPI spec only mentions RAML DataType [here](https://github.com/asyncapi/asyncapi/blob/master/versions/2.0.0/asyncapi.md#schema-formats-table).

**Related issue(s)**
Fixes #11

### CODE REVIEW

**Comment by fmvilas** [comment]
I think we can't just add them as regular tests. It would be unfair to say that a given parser doesn't have 100% AsyncAPI support because it's not supporting optional features. We should probably consider adding the ability to have optional tests or separate sets of tests.

**Comment by jstoiko** [comment]
@postatum: let’s find a way to make those tests “optional”. In the meantime, I would suggest to take out all the non-JSON schema tests out of this PR (in another PR) and make this PR “just” about what’s mandatory.

**Comment by postatum** [comment]
> @postatum: let’s find a way to make those tests “optional”.

My initial thought is to add another naming convention similar to those we have already. E.g. tck tests which have "optional" in their name won't affect coverage percentage.

> In the meantime, I would suggest to take out all the non-JSON schema tests out of this PR (in another PR) and make this PR “just” about what’s mandatory.

Done: https://github.com/asyncapi/tck/pull/13

**Comment by postatum** [comment]
This PR can be reviewed. I've reworked it to use new optional tests naming convention.

File: tests/asyncapi-2.0/Message Object/Payloads/includes/user-datatype.raml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#%RAML 1.0 DataType
+
+properties:
+  username:
+    type: string
```
Comment (jstoiko) [inline tests/asyncapi-2.0/Message Object/Payloads/includes/user-datatype.raml:5]: `type: string` and `required: true` are implicit in RAML so I would say let's remove those lines

Comment (postatum) [reply tests/asyncapi-2.0/Message Object/Payloads/includes/user-datatype.raml]: Done

**Comment by jstoiko** [comment]
> I didn't add RAML Library tests because AsyncAPI spec only mentions RAML DataType here.

Libraries are fine. ["RAML data type" refers to RAML's type system as a whole](https://github.com/raml-org/raml-spec/blob/master/versions/raml-10/raml-10.md/#raml-data-types), it's not limited to RAML "DataType" fragment. 

That being said, it would be nice to have TCK tests that leverage a RAML library as well as a RAML API (root) definition with `types:` and refer to some of these types using [JSON references](https://github.com/asyncapi/asyncapi/blob/master/versions/2.0.0/asyncapi.md#reference-object).

**Comment by postatum** [comment]
> > I didn't add RAML Library tests because AsyncAPI spec only mentions RAML DataType here.
> 
> Libraries are fine. ["RAML data type" refers to RAML's type system as a whole](https://github.com/raml-org/raml-spec/blob/master/versions/raml-10/raml-10.md/#raml-data-types), it's not limited to RAML "DataType" fragment.
> 
> That being said, it would be nice to have TCK tests that leverage a RAML library as well as a RAML API (root) definition with `types:` and refer to some of these types using [JSON references](https://github.com/asyncapi/asyncapi/blob/master/versions/2.0.0/asyncapi.md#reference-object).

Done

**Pull request merged**